src/backend/optimizer/plan/aqumv.c

```diff
@@ -468,17 +429,16 @@ answer_query_using_materialized_views(PlannerInfo *root,
 			/*
 			 * AQUMV_FIXME_MVP
 			 * Use new query's ecs.
-			 * Equivalence Class is not supported now, we may lost some ECs if the mv_query has
+			 * Equivalence Class is not supported now, we may lost some ECs if the viewQuery has
 			 * equal quals or implicit ones.
 			 * But keeping them also introduces more complex as we should process them like target list.
 			 * Another flaw: the generated Filter expressions by keeping them are pointless as all
 			 * rows of mv have matched the filter expressions.
 			 * See more in README.cbdb.aqumv
 			 */
 			root->eq_classes = subroot->eq_classes;
-			/* Replace relation, don't close the right one. */
 			current_rel = mv_final_rel;
-			table_close(matviewRel, AccessShareLock);
+			table_close(matviewRel, NoLock);
```

Comment:
also line 164:
```
if (need_close)
			table_close(matviewRel, AccessShareLock);
```

Reply:
When need_close is true, we don't need to hold AccessShareLock there because that matviewRel is abandoned by  us.